### DISCUSSION TITLE
Drive: Adding Custom Service (Azure)

### DISCUSSION THREAD

**Discussion by Tob0t** [adonisjs/core#4708] [Ideas]
Hi,
I would like to use Drive to work with Azure Blob Storage. Until Adonis v5, I used the package https://github.com/AlexanderYW/Adonis-Drive-Azure-Storage, which does not seem to work anymore since you can only add items from a predefined list (fs, s3, gcs).
Any chance to enable adding a custom service as well in the near future?

FYI: Flydrive is already supporting it: https://flydrive.dev/docs/advanced/custom_drivers

**Comment by necessarylion** [comment]
I was looking for the same thing, and I end up creating custom driver here.
https://github.com/necessarylion/flydrive-azure-driver
I would appreciate any input or help.

**Reply by thetutlage** [reply]
@necessarylion Thanks for creating and sharing it. It will be great if you can submit this driver on https://packages.adonisjs.com/ for better discoverability.

**Reply by necessarylion** [reply]
Thanks! I have submitted a PR here https://github.com/adonisjs-community/adonis-packages/pull/116
